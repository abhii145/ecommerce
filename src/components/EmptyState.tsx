import { ReactNode } from "react"

interface EmptyStateProps {
  message: string
  buttonText: string
  onButtonClick: () => void
  icon: ReactNode
}

const EmptyState = ({
  message,
  buttonText,
  onButtonClick,
  icon,
}: EmptyStateProps) => (
  <div className="flex flex-col items-center justify-center mt-12 space-y-4">
    <div className="text-6xl text-gray-300" aria-hidden="true">
      {icon}
    </div>
    <p className="text-lg text-gray-500" role="alert">
      {message}
    </p>
    <button
      className="bg-blue-500 text-white py-2 px-6 rounded-md shadow-md hover:bg-blue-600 transition-transform duration-300 hover:scale-105 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:ring-opacity-50"
      onClick={onButtonClick}
      aria-label={buttonText}
    >
      {buttonText}
    </button>
  </div>
)

export default EmptyState
