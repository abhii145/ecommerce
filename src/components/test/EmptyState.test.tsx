import React from "react"
import { render, screen, fireEvent } from "@testing-library/react"
import EmptyState from "../EmptyState"

describe("EmptyState", () => {
  const mockOnButtonClick = jest.fn()
  const props = {
    message: "No items found",
    buttonText: "Add Item",
    onButtonClick: mockOnButtonClick,
    icon: (
      <span role="img" aria-label="icon">
        üîç
      </span>
    ),
  }

  it("matches snapshot", () => {
    const { asFragment } = render(<EmptyState {...props} />)

    expect(asFragment()).toMatchSnapshot()
  })

  it("renders correctly with all elements", () => {
    render(<EmptyState {...props} />)

    expect(screen.getByLabelText("icon")).toBeInTheDocument()
    expect(screen.getByRole("alert")).toHaveTextContent("No items found")
    expect(screen.getByRole("button", { name: "Add Item" })).toBeInTheDocument()
  })

  it("calls onButtonClick when button is clicked", () => {
    render(<EmptyState {...props} />)

    fireEvent.click(screen.getByRole("button", { name: "Add Item" }))
    expect(mockOnButtonClick).toHaveBeenCalled()
  })
})
