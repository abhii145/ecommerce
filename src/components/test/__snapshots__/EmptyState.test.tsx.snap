// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col items-center justify-center mt-12 space-y-4"
  >
    <div
      aria-hidden="true"
      class="text-6xl text-gray-300"
    >
      <span
        aria-label="icon"
        role="img"
      >
        üîç
      </span>
    </div>
    <p
      class="text-lg text-gray-500"
      role="alert"
    >
      No items found
    </p>
    <button
      aria-label="Add Item"
      class="bg-blue-500 text-white py-2 px-6 rounded-md shadow-md hover:bg-blue-600 transition-transform duration-300 hover:scale-105 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:ring-opacity-50"
    >
      Add Item
    </button>
  </div>
</DocumentFragment>
`;
